src/views/Charts.vue

```diff
@@ -0,0 +1,224 @@
+  isCandlestickType = false;
+  // Set XOR asset as default
+  prices: Array<[string, any]> = [];
+  currentAsset: Asset = KnownAssets.get(KnownSymbols.XOR);
+  chartSeriesData: number[] = [];
+  chartXAxisData: string[] = [];
+  candleChartXAxisData: string[] = [];
+  candleStickSeriesData: Array<Array<number>> = [];
+  candleTimeSteps = [4, 8, 12, 24];
+  candleTimeStep = this.candleTimeSteps[0];
+
+  yAxeLimits = {
+    min: 45,
+    max: 70,
+  };
+
+  @Watch('candleTimeStep')
+  private handleChangeTimeStep(value: number): void {
+    this.updateCandleGraph();
+  }
+
+  get chartTitle(): string {
+    return this.t('charts.chartTitle', { assetSymbol: this.currentAsset?.symbol });
+  }
+
+  get chartData(): any {
+    return {
+      title: {
+        text: this.chartTitle,
+      },
+      xAxis: {
+        type: 'category',
+        name: this.t('charts.date'),
+        data: this.chartXAxisData,
+      },
+      yAxis: {
+        type: 'value',
+        min: this.yAxeLimits.min,
+        max: this.yAxeLimits.max,
+      },
+      series: [
+        {
+          type: 'line',
+          data: this.chartSeriesData,
+        },
+      ],
+    };
+  }
+
+  get optionCandleStick(): any {
+    return {
+      title: {
+        text: this.chartTitle,
+      },
+      xAxis: {
+        data: this.candleChartXAxisData,
+      },
+      yAxis: {
+        min: this.yAxeLimits.min,
+        max: this.yAxeLimits.max,
+      },
+      series: [
+        {
+          type: 'candlestick',
+          data: this.candleStickSeriesData,
+        },
+      ],
+    };
+  }
+
+  created() {
+    this.withApi(async () => {
+      try {
+        const data = await SubqueryExplorerService.getHistoricalPriceForAsset(this.currentAsset.address, 100);
+        if (data) {
+          this.prices = Object.entries(data).reverse();
+          // Sort prices to identify min and max yAxe values
+          const sortedPrices = [...this.prices];
+          sortedPrices.sort((a, b) => {
+            return +a[1] - +b[1];
+          });
+
+          this.yAxeLimits.min = Math.round(+this.getFPNumberFromCodec(sortedPrices[0][1])) - 1;
+          this.yAxeLimits.max = Math.round(+this.getFPNumberFromCodec(sortedPrices[sortedPrices.length - 1][1])) + 1;
+
+          this.updateGraphs();
+          this.updateCandleGraph();
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  }
+
+  updateGraphs(): void {
+    const closeIndex = this.candleTimeStep - 1;
+    for (let i = 0; i < this.prices.length; i++) {
+      const date = new Date(+this.prices[i][0]);
+      this.chartXAxisData.push(`
+        ${date.getUTCDate()}/${this.formatDateItem(date.getUTCMonth() + 1)}
+        ${this.formatDateItem(date.getUTCHours())}:${this.formatDateItem(date.getUTCMinutes())}
+      `);
+      this.chartSeriesData.push(this.formatPrice(this.prices[i][1]));
+      this.updateCandleGraph();
```

Comment:
As i understand this method prepares data for linear chart. Do we need this call `this.updateCandleGraph()` here?

Reply:
Refactored this place, thank you!